tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BCompileUtil.java

```diff
@@ -570,30 +570,20 @@ private static CompileResult compileOnJBallerina(CompilerContext context, String
         BLangPackage bLangPackage = (BLangPackage) compileResult.getAST();
         byte[] compiledJar = JVMCodeGen.generateJarBinary(false, bLangPackage, context, packageName);
         JBallerinaInMemoryClassLoader classLoader = new JBallerinaInMemoryClassLoader(compiledJar);
-        String entryClassName = FileUtils.cleanupFileExtension(packageName).replace('.', '_');
-        entryClassName = getQualifiedClassName(bLangPackage.packageID.orgName.value, packageName, entryClassName);
-        Class<?> clazz = classLoader.loadClass(entryClassName);
-
-        // invoke the init function
+        String initClassName = BFileUtil.getQualifiedClassName(bLangPackage.packageID.orgName.value,
+                                                               packageName, MODULE_INIT_CLASS_NAME);
+        Class<?> initClazz = classLoader.loadClass(initClassName);
         String funcName = cleanupFunctionName(((BLangPackage) compileResult.getAST()).initFunction);
         try {
-            Method method = clazz.getDeclaredMethod(funcName, Strand.class);
+            Method method = initClazz.getDeclaredMethod(funcName, Strand.class);
             method.invoke(null, new Strand(new Scheduler()));
         } catch (Exception e) {
             throw new RuntimeException("Error while invoking function '" + funcName + "'", e);
         }
-
-        compileResult.setEntryClass(clazz);
+        compileResult.setClassLoader(classLoader);
```

Comment:
Previously we had only one class file. Hence string entry class will be ok. But now when we have multiple class files and during test we need to load the classes on demand ( multiple entry points) . Hence I am storing class loader . If not static values created during init method will be lost.